spectrometer.py

```diff
@@ -1,303 +1,331 @@
-
-for wl in range(400,1001,50):
-	x=xd-(wl/wavelengthFactor)
-	y0=math.tan(spectrumAngle)*(xd-x)+aperture['y']
-	draw.line((x,y0+h+5, x,y0+h-5))
-	draw.text((x,y0+h+15),str(wl))
-	
-exposure=maxResult/(255+255+255)
-print "ideal exposure between 0.15 and 0.30"
-print "exposure=",exposure
-if (exposure<0.15):
-	print "consider increasing shutter time"
-elif (exposure>0.3):
-	print "consider reducing shutter time"
-
-#save image with markup
-outputFilename=name+"_out.jpg"
-ImageFile.MAXBLOCK = 2**20
-im.save(outputFilename, "JPEG", quality=80, optimize=True, progressive=True)
-
-#normalise results
-for wavelength in results:
-	results[wavelength]=results[wavelength]/maxResult 
-
-#save csv of results
-csvFilename=name+".csv"
-csv = open(csvFilename, 'w')
-csv.write("wavelength,amplitude\n")
-for wavelength in results:
-	csv.write(wavelength)
-	csv.write(",")
-	csv.write("{:0.3f}".format(results[wavelength]))
-	csv.write("\n");
-csv.close()
-
-#generate spectrum diagram
-antialias=4
-w=600*antialias
-h2=300*antialias
-
-
-h=h2-20*antialias
-sd = Image.new('RGB', (w, h2), (255,255,255))
-draw = ImageDraw.Draw(sd)
-
-w1= 380.0
-w2= 780.0
-f1 = 1.0/w1;
-f2 = 1.0/w2;
-for x in range(0,w,1):
-	# Iterate across frequencies, not wavelengths
-	lambda2 = 1.0/(f1-(float(x)/float(w)*(f1-f2)))
-	c=wavelengthToColor(lambda2)
-	draw.line((x, 0, x, h),fill=c)
-	
-pl=[]
-pl.append( (w,0) )
-pl.append( (w,h) )
-for wavelength in results:
-	wl=float(wavelength)
-	x=int( (wl-w1)/(w2-w1) * w )
-	#print wavelength,x
-	pl.append( ( int(x), int((1-results[wavelength])*h) ) )
-pl.append( (0,h) )
-pl.append( (0,0) )
-draw.polygon(pl,fill="#FFF")
-draw.polygon(pl)
-
-font = ImageFont.truetype("Lato-Regular.ttf", 12*antialias)
-draw.line((0,h,w,h),fill="#000",width=antialias)
-
-for wl in range(400,1001,10):
-	x=int( (float(wl)-w1)/(w2-w1) * w )
-	draw.line((x,h, x,h+3*antialias),fill="#000",width=antialias)
-
-
-for wl in range(400,1001,50):
-	x=int( (float(wl)-w1)/(w2-w1) * w )
-	draw.line((x,h, x,h+5*antialias),fill="#000",width=antialias)
-	wls=str(wl)
-	tx=draw.textsize(wls,font=font)
-	draw.text((x-tx[0]/2,h+5*antialias),wls,font=font)
-
-#save chart
-sd = sd.resize((w/antialias,h/antialias), Image.ANTIALIAS)
-outputFilename=name+"_chart.png"
-sd.save(outputFilename, "PNG", quality=95, optimize=True, progressive=True)
+    # Let the intensity fall off near the vision limits
+    if 380 <= lambda2 < 420:
+        factor = 0.2 + 0.8 * (lambda2 - 380) / (420 - 380)
+    elif 600 <= lambda2 < 780:
+        factor = 0.2 + 0.8 * (780 - lambda2) / (780 - 600)
+    return int(255 * color[0] * factor), int(255 * color[1] * factor), int(255 * color[2] * factor)
+
+
+def take_picture(camera, name, shutter):
+    camera.vflip = True
+    camera.framerate = Fraction(1, 6)
+    camera.shutter_speed = shutter
+    camera.iso = 100
+    camera.exposure_mode = 'off'
+    camera.awb_mode = 'off'
+    camera.awb_gains = (1, 1)
+
+    time.sleep(3)
+
+    raw_filename = name + "_raw.jpg"
```

Comment:
Similarly, perhaps `name` is better named something like `filename` or `filename_prefix`.